deploy/circle_deploy.sh

```diff
@@ -39,7 +39,7 @@ then
 	CF_APP="cg-dashboard-demo"
 else
   echo Unknown environment, quitting. >&2
-  exit 1
+  exit 0
```

Comment:
The old logic only called the deploy script if it had one of the appropriate branches or a matching tag.

To replicate that same behavior with 2.0, I'd have to use a filter statement, which would either be in the workflow, as a deploy step, or in the jobs. That would double the number of jobs/steps, so each job would be held up by 22.9Gb Docker install, and we'd have to make sure we'd cached and restored all the correct paths between the jobs.

Since the deploy script has the logic to deploy to the correct place, there's no harm in having it called with each job, but just exit 0 (OK) if there's no matching tag/branch.

I know this whole PR is ugly as sin, so I hope we tamp down our aesthetic inclinations and move it forward.